File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    static class Process {
        int idx, v;
        Process(int idx, int v){
            this.idx = idx;
            this.v = v;
        }
    }
    public int solution(int[] priorities, int location) {
        PriorityQueue<Process> pq = new PriorityQueue<>((a, b) -> {
            return b.v - a.v;
        });
        
        Queue<Process> q = new ArrayDeque<>();
        for (int i = 0; i < priorities.length; i++) {
            Process p = new Process(i, priorities[i]);
            pq.offer(p);
            q.offer(p);
        }
        
        int answer = 1;
        while (!q.isEmpty()) {
            Process p = q.poll();
            
            if (pq.peek().v == p.v) {
                pq.poll();
                
                if(p.idx == location) return answer;
                
                answer++;
            }
            else {
                
                q.offer(p);
                continue;
            }
            

        }
        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    
    
    const q = [];
    
    for (let i = 0; i < priorities.length; i++) {
        q.push({no:i, v:priorities[i]});
    }
    
    const arr = [];
    let cnt = 1;
    while(q.length > 0) {
        let max = q[0].v;
        let idx = 0;
        for (let i = 1; i < q.length; i++) {
            if (max < q[i].v) {
                max = q[i].v;
                idx = i;
            }
        }
        
        for (let i = 0; i < idx; i++) {
            q.push(q.shift());
        }
        
        arr[q.shift().no] = cnt;
        cnt++;
    }
    console.log(arr);

    return arr[location];
}